hummingbot/connector/parrot.py

```diff
@@ -55,44 +60,65 @@ async def get_campaign_summary(exchange: str, trading_pairs: List[str] = []) ->
     return results
 
 
-async def get_market_snapshots(market_id: int):
+async def get_market_snapshots(market_id: int, timestamp: int = 1657747860000):
```

Comment:
This is very bad, I saw the method the timestamp is hardcoded and you are just calling the method without any timestamp when this is used

Reply:
Agreed, I must have wanted to update this with time() and forgot after testing it

Reply:
The reason for a default value:
```
{'reason': "Missing params in payload, all params {'market_id', 'timestamp'} "
           'must be present and valid.'}
```
when using '-1':
`{'message': 'Data not available for timestamp -1.', 'status': 'error'}`

Reply:
Ok, finally figured it out, the valid timestamps can be gathered from the charts, then it can be used for the snapshots. Refactoring and adding the appropriate tests should be completed tomorrow

Reply:
This is still harcoded, I think that you have to use the timestamp that it's provided in the first query that you make to retrieve the campaigns. There is one value that is last_snapshot_ts